### PR TITLE
Heal 31, Heal 32, Heal 75, Heal 77

### PR DESCRIPTION
**Pull request by aishwaryaravi19**
Screenshot 2023-04-11 at 11 05 42 PM
Screenshot 2023-04-09 at 7 14 58 PM
Screenshot 2023-04-11 at 11 05 53 PM

### CODE REVIEW

File: client/src/components/Doctor/DoctorDashboard.tsx [hunk 1]
```diff
@@ -1,35 +1,164 @@
...
+    //     hash.update(fileBuffer);
+    //     const hashDigest = hash.digest('hex');
+    //     resolve(hashDigest);
+    //     };
+    //     fileReader.onerror = () => {
+    //     reject(fileReader.error);
+    //     };
+    //     fileReader.readAsArrayBuffer(file);
+    // });
+    // }
```
Comment (varunalla) [inline client/src/components/Doctor/DoctorDashboard.tsx]: remove if unnecessary. The file upload functionality is not implemented

File: client/src/components/Doctor/DoctorDashboard.tsx [hunk 2]
```diff
@@ -1,35 +1,164 @@
...
+    const navigate = useNavigate();
+    //const [contract, setContract] = useState<any>();
+    const [file, setFile] = useState<File | null>(null);
+
     const logouthandler = () => {
         logout?.();
         navigate('/doctorlogin');
     }
+
+    const handleFileUpload = (event: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (varunalla) [inline client/src/components/Doctor/DoctorDashboard.tsx]: encrypt files being uploaded to s3 as doctor credentials are confidential

File: client/src/components/HealthCareProvider/HCProviderDashboard.tsx [hunk 3]
```diff
@@ -1,28 +1,78 @@
...
+              </tr>
+            </thead>
+            <tbody className="divide-y divide-gray-200">
+              {verificationRequests && verificationRequests.map((request: { doctorName: string | number | boolean | React.ReactElement<any, string | React.JSXElementConstructor<any>> | React.ReactFragment | React.ReactPortal | null | undefined; credentialsHash: string | number | boolean | React.ReactElement<any, string | React.JSXElementConstructor<any>> | React.ReactFragment | React.ReactPortal | null | undefined; approved: any; }, index) => (
+                <tr key={index}>
+                  <td className="px-6 py-4 text-center">{request.doctorName}</td>
+                  <td className="px-6 py-4 text-center">{request.credentialsHash}</td>
+                  <td className="px-6 py-4 text-center">{request.approved ? "Approved" : "Pending"}</td>
+                  <td className="px-6 py-4 text-center">
+                    {!request.approved && (
```
Comment (varunalla) [inline client/src/components/HealthCareProvider/HCProviderDashboard.tsx]: if request.approved is false, there is no distinction between pending and rejected requests. this seems to be a bug.

File: client/src/providers/HealthProvider.tsx [hunk 4]
```diff
@@ -1,41 +1,43 @@
...
-export type Patient = {
-  name: string;
-  age: number;
-  email: string;
-};
+interface Request {
+    doctor: string;
+    doctorName: string;
+    credentialsHash: string;
+    approved: boolean;
```
Comment (varunalla) [inline client/src/providers/HealthProvider.tsx]: if request.approved is false, there is no distinction between pending and rejected requests. this seems to be a bug. how will the request be associated with health care provider?

File: client/src/providers/HealthProvider.tsx [hunk 5]
```diff
@@ -131,47 +139,112 @@ export const HealthProvider: React.FC<Props> = ({ children, ...props }) => {
...
-      setError('Error Loading Health Contract');
+            console.log(contract);
+            const register = await contract.registerHCProvider(name, email, address, phone);
+            register.wait();
+            console.log(register);
+        }
+        catch (err) {
+            console.log(err)
+            setError("Error Loading Health Contract")
+        }
```
Comment (varunalla) [inline client/src/providers/HealthProvider.tsx]: install prettier and make sure this is formatted properly

File: client/src/providers/HealthProvider.tsx [hunk 6]
```diff
@@ -131,47 +139,112 @@ export const HealthProvider: React.FC<Props> = ({ children, ...props }) => {
...
-        registerDoctorHealthBlockContract,
-        registerHCProviderHealthBlockContract,
-      }}
-    >
-      {children}
-    </HealthContext.Provider>
-  );
+    const handleRaiseRequest = async (doctorName: string, file: File | null) => {
+        if (!file) return;
+        console.log("doctor name", doctorName)
```
Comment (varunalla) [inline client/src/providers/HealthProvider.tsx]: please remove all console.log in which ever section you see them.

File: client/src/providers/HealthProvider.tsx [hunk 7]
```diff
@@ -131,47 +139,112 @@ export const HealthProvider: React.FC<Props> = ({ children, ...props }) => {
...
+    const handleApproveRequest = async (requestId: number) => {
+        try {
+          const web3modal = new Web3Modal();
+          const connection = await web3modal.connect();
+          const provider = new ethers.providers.Web3Provider(connection);
+          const signer = provider.getSigner();
+          const contract = await fetchContract(signer);
+          const tx = await contract.approveRequest(requestId);
+          await tx.wait();
+          fetchRequests();
```
Comment (varunalla) [inline client/src/providers/HealthProvider.tsx]: is this call necessary? can you explain why?

File: config/constants.js [hunk 8]
```diff
@@ -1,6 +1,5 @@
-const healthBlockAddress =
-  process.env.HEALTHBLOCK_ADDRESS ||
-  "0x6Fd2B45F4Eff401654028c5097fdE9c6a8f07298";
-module.exports = {
-  healthBlockAddress,
-};
+const healthBlockAddress=process.env.HEALTHBLOCK_ADDRESS || "0x2c1d88575485Cc7745Eb38CD95FC3818e928476c";
+console.log(healthBlockAddress);
```
Comment (varunalla) [inline config/constants.js:4]: remove console.log and fix formatting

**Review by varunalla — CHANGES_REQUESTED** [review]
Please delete contents of client/src/contracts folder. commit it and run type check and  checkin those files.

File: client/src/providers/HealthProvider.tsx (hunk 7 above)
Comment (aishwaryaravi19) [reply client/src/providers/HealthProvider.tsx]: yes, it is necessary to set verification request after each approval/rejection

File: client/src/components/Doctor/DoctorDashboard.tsx (hunk 2 above)
Comment (aishwaryaravi19) [reply client/src/components/Doctor/DoctorDashboard.tsx]: It will be addressed as part of HEAL 83 & 84

**Review by varunalla — APPROVED** [review]
approved! please test once merged!

**Pull request merged**